In [1]:
import pandas as pd
import numpy as np

In [2]:
o=pd.read_csv('/content/orders.csv')
ws=pd.read_csv('/content/website_sessions.csv')
wp=pd.read_csv('/content/website_pageviews.csv')

In [3]:
o.head()

,order_id,created_at,website_session_id,user_id,primary_product_id,items_purchased,price_usd,cogs_usd
0,1,2012-03-19 10:42:46,20,20,1,1,49.99,19.49
1,2,2012-03-19 19:27:37,104,104,1,1,49.99,19.49
2,3,2012-03-20 06:44:45,147,147,1,1,49.99,19.49
3,4,2012-03-20 09:41:45,160,160,1,1,49.99,19.49
4,5,2012-03-20 11:28:15,177,177,1,1,49.99,19.49


In [4]:
ws.head()


,website_session_id,created_at,user_id,is_repeat_session,utm_source,utm_campaign,utm_content,device_type,http_referer
0,1,2012-03-19 08:04:16,1,0,gsearch,nonbrand,g_ad_1,mobile,https://www.gsearch.com
1,2,2012-03-19 08:16:49,2,0,gsearch,nonbrand,g_ad_1,desktop,https://www.gsearch.com
2,3,2012-03-19 08:26:55,3,0,gsearch,nonbrand,g_ad_1,desktop,https://www.gsearch.com
3,4,2012-03-19 08:37:33,4,0,gsearch,nonbrand,g_ad_1,desktop,https://www.gsearch.com
4,5,2012-03-19 09:00:55,5,0,gsearch,nonbrand,g_ad_1,mobile,https://www.gsearch.com


In [6]:
wp.head()

,website_pageview_id,created_at,website_session_id,pageview_url
0,1,2012-03-19 08:04:16,1,/home
1,2,2012-03-19 08:16:49,2,/home
2,3,2012-03-19 08:26:55,3,/home
3,4,2012-03-19 08:37:33,4,/home
4,5,2012-03-19 09:00:55,5,/home


In [7]:
o.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32313 entries, 0 to 32312
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   order_id            32313 non-null  int64  
 1   created_at          32313 non-null  object 
 2   website_session_id  32313 non-null  int64  
 3   user_id             32313 non-null  int64  
 4   primary_product_id  32313 non-null  int64  
 5   items_purchased     32313 non-null  int64  
 6   price_usd           32313 non-null  float64
 7   cogs_usd            32313 non-null  float64
dtypes: float64(2), int64(5), object(1)
memory usage: 2.0+ MB


In [8]:
ws.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 472871 entries, 0 to 472870
Data columns (total 9 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   website_session_id  472871 non-null  int64 
 1   created_at          472871 non-null  object
 2   user_id             472871 non-null  int64 
 3   is_repeat_session   472871 non-null  int64 
 4   utm_source          389543 non-null  object
 5   utm_campaign        389543 non-null  object
 6   utm_content         389543 non-null  object
 7   device_type         472871 non-null  object
 8   http_referer        432954 non-null  object
dtypes: int64(3), object(6)
memory usage: 32.5+ MB


In [9]:
wp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1188124 entries, 0 to 1188123
Data columns (total 4 columns):
 #   Column               Non-Null Count    Dtype 
---  ------               --------------    ----- 
 0   website_pageview_id  1188124 non-null  int64 
 1   created_at           1188124 non-null  object
 2   website_session_id   1188124 non-null  int64 
 3   pageview_url         1188124 non-null  object
dtypes: int64(2), object(2)
memory usage: 36.3+ MB


In [10]:
ws.isnull().sum()

,0
website_session_id,0
created_at,0
user_id,0
is_repeat_session,0
utm_source,83328
utm_campaign,83328
utm_content,83328
device_type,0
http_referer,39917


In [14]:
ws[ws['utm_source'].isnull() & ~ws['website_session_id'].isin(o['website_session_id'].unique())]

,website_session_id,created_at,user_id,is_repeat_session,utm_source,utm_campaign,utm_content,device_type,http_referer
904,905,2012-03-25 07:24:18,905,0,NaN,NaN,NaN,mobile,NaN
914,915,2012-03-25 10:45:03,915,0,NaN,NaN,NaN,mobile,NaN
933,934,2012-03-25 14:23:48,934,0,NaN,NaN,NaN,desktop,NaN
1044,1045,2012-03-26 11:54:49,846,1,NaN,NaN,NaN,desktop,NaN
1050,1051,2012-03-26 12:48:41,152,1,NaN,NaN,NaN,desktop,NaN
...,...,...,...,...,...,...,...,...,...
472835,472836,2015-03-19 06:25:38,394289,0,NaN,NaN,NaN,desktop,https://www.gsearch.com
472846,472847,2015-03-19 06:57:13,378178,1,NaN,NaN,NaN,mobile,https://www.gsearch.com
472860,472861,2015-03-19 07:38:28,383375,1,NaN,NaN,NaN,desktop,NaN
472863,472864,2015-03-19 07:41:11,394311,0,NaN,NaN,NaN,desktop,https://www.gsearch.com


In [15]:
ws.drop(ws[ws['utm_source'].isnull() & ~ws['website_session_id'].isin(o['website_session_id'].unique())].index, inplace=True)

**77210 sessions rows dont have campaign details and orders too then we droped it.**

In [23]:
ws.shape

(395661, 9)

In [17]:
ws.isnull().sum()

,0
website_session_id,0
created_at,0
user_id,0
is_repeat_session,0
utm_source,6118
utm_campaign,6118
utm_content,6118
device_type,0
http_referer,2856


**not deleting 6118 null rows because they have their orders data in orders table**

In [19]:
wp.shape

(1188124, 4)

In [18]:
wp[~wp['website_session_id'].isin(ws['website_session_id'].unique())]

,website_pageview_id,created_at,website_session_id,pageview_url
1713,1714,2012-03-25 07:24:18,905,/home
1715,1716,2012-03-25 07:28:38,905,/products
1738,1739,2012-03-25 10:45:03,915,/home
1739,1740,2012-03-25 10:48:08,915,/products
1771,1772,2012-03-25 14:23:48,934,/home
...,...,...,...,...
1188098,1188099,2015-03-19 07:42:03,472861,/products
1188103,1188104,2015-03-19 07:43:37,472861,/the-original-mr-fuzzy
1188106,1188107,2015-03-19 07:45:31,472864,/products
1188109,1188110,2015-03-19 07:48:12,472864,/the-original-mr-fuzzy


In [20]:
wp.drop(wp[~wp['website_session_id'].isin(ws['website_session_id'].unique())].index,inplace=True)

**deleted all pageview rows dont have their sessions and orders details available**

In [21]:
wp.shape

(1010875, 4)

In [25]:
o.to_csv('orders.csv', index=False)
ws.to_csv('website_sessions.csv', index=False)
wp.to_csv('website_pageviews.csv', index=False)